File: Command/python-source/cookery/CookPool.py
from __future__ import annotations

from .Bartender import Bartender
from .Cook import Cook


class CookPool:
    """Отдаёт исполнителей. В настоящей системе на его месте был бы пул."""

    _cook = Cook()
    _bartender = Bartender()

    @classmethod
    def get_cook(cls) -> Cook:
        return cls._cook

    @classmethod
    def get_bartender(cls) -> Bartender:
        return cls._bartender

File: JavaSource/src/main/java/ru/akhcheck/patterns/command/cookery/CookPool.java
package ru.akhcheck.patterns.command.cookery;

/** Отдаёт исполнителей. В настоящей системе на его месте был бы пул. */
public final class CookPool {
    private static final Cook COOK = new Cook();
    private static final Bartender BARTENDER = new Bartender();

    private CookPool() {
    }

    public static Cook getCook() {
        return COOK;
    }

    public static Bartender getBartender() {
        return BARTENDER;
    }
}
